module/zfs/spa.c

```diff
@@ -4346,6 +4359,7 @@ spa_load_impl(spa_t *spa, spa_import_type_t type, char **ereport)
 		spa_config_exit(spa, SCL_CONFIG, FTAG);
 	}
 
+	spa_import_progress_remove(spa_guid(spa));
```

Comment:
Is there value in not removing the kstat entry until the pool is exported?

Reply:
We discussed this offline, and decided not to keep entries after the import was finished.

The logic was based on what data one might get from the entry after the fact:
* successful import
** MMP check time - this may be different next time, since it is based on I/O latency.
** txg rewound to during recovery - this should be retained permanently, probably via zpool history
** total import time - the time visible from this procfile is just the tryimport or just the real import, not both, and in both cases not including the userspace time.  One could get total time, likely what the user wants, by recording it in userspace.
* failed import
** MMP check time - one could just start the import again, read the time, and then cancel.  This is likely what the admin would think to do anyway.
** txg rewound to which failed - this is an edge case, and something that the admin would be doing by hand.  So if they want to know this, they will likely be monitoring the procfile during the import anyway.